spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -177,19 +181,44 @@ public XDStreamParser(ModuleRegistry moduleRegistry,
 		// And while we're at it (and type is known), validate module name and options
 		List<ModuleDescriptor> result = new ArrayList<ModuleDescriptor>(builders.size());
 		for (ModuleDescriptor.Builder builder : builders) {
-			builder.setType(determineType(builder, builders.size() - 1, parsingContext));
-
-			// definition is guaranteed to be non-null here
-			ModuleDefinition moduleDefinition = moduleRegistry.findDefinition(builder.getModuleName(), builder.getType());
-			builder.setModuleDefinition(moduleDefinition);
-			ModuleOptionsMetadata optionsMetadata = moduleOptionsMetadataResolver.resolve(moduleDefinition);
-			if (parsingContext.shouldBindAndValidate()) {
-				try {
-					optionsMetadata.interpolate(builder.getParameters());
+			ModuleType moduleType;
+			try {
+				moduleType = determineType(builder, builders.size() - 1, parsingContext);
+			}
+			catch (NoSuchModuleException nsme) {
+				if (errorAccumulator != null) {
+					errorAccumulator.add(nsme);
+					// 'processor' below effectively indicates 'unknown' - a caller passing an
+					// error accumulator should be aware that this can happen (the accumulator
+					// will contain the exception that indicates it did)
+					moduleType = ModuleType.processor;
```

Comment:
This seems the simplest option instead of adding lots of null handling throughout the following methods that get called.